lib/views/desktop/send/widgets/DTSelectOrDropAFile.dart

```diff
@@ -21,15 +23,30 @@ class DTSelectOrDropAFile extends StatefulWidget {
 class _DTSelectOrDropAFile extends State<DTSelectOrDropAFile> {
   @override
   Widget build(BuildContext context) {
+    bool flag = false;
     return Container(
       color: widget.dragEntered
           ? Color(0xff3A2655)
           : Theme.of(context).dialogBackgroundColor,
       child: DropTarget(
         onDragDone: (detail) async {
-          await widget.onFileDropped(detail.files.first.readOnlyFile());
+          if (flag == true) return;
+          try {
+            File file = detail.files.first.readOnlyFile();
+            await widget.onFileDropped(file);
+            flag = true;
```

Comment:
This and the `flag = true` in the catch block can be removed, because both branches set `flag`. Maybe right after the conditional that exits the function when `flag` is set you can place that assignment?